core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -457,4 +457,15 @@ public void writeNamedWriteable(NamedWriteable namedWriteable) throws IOExceptio
         writeString(namedWriteable.getName());
         namedWriteable.writeTo(this);
     }
+
+    /**
+     * Writes a list of {@link NamedWriteable} to the current stream, by first writing its size and then iterating over the objects
+     * in the list
+     */
+    public void writeNamedWritableList(List<? extends NamedWriteable> list) throws IOException {
```

Comment:
thinking out loud, should we support Collection instead of List in both methods?